src/Block/NonceFieldBlock.php

```diff
@@ -36,7 +41,18 @@ public function __construct(
     ) {
         $this->_setNonce($nonce)
             ->_setFieldName($fieldName)
-            ->_setUseReferer($useReferer);
+            ->_setUseReferer($useReferer)
+            ->_setRefererFieldName(static::REFERER_FIELD_NAME);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    protected function _getRefererUrl()
+    {
+        return \esc_attr(\wp_unslash($_SERVER['REQUEST_URI']));
```

Comment:
@XedinUnknown My bad. I copy-pasted the entire thing out of WordPress without much thought. I'll remove the `esc_attr` call and make sure it is used during rendering.

Regarding the referer URL, I would tend towards injecting it.